File: src/main/java/com/practica/crean/arquitecture/core/usecases/impl/GuardarEstudianteImpl.java
package com.practica.crean.arquitecture.core.usecases.impl;

import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.core.port.EstudianteRepository;
import com.practica.crean.arquitecture.core.usecases.GuardarEstudiante;

public class GuardarEstudianteImpl implements GuardarEstudiante {

    private EstudianteRepository repository;

    public GuardarEstudianteImpl(EstudianteRepository repository) {
        this.repository = repository;
    }

    @Override
    public Estudiante execute(Estudiante estudiante) {
        return repository.guardarEstudiante(estudiante);
    }

}

File: src/main/kotlin/com/practica/crean/arquitecture/core/usecases/impl/GuardarEstudianteImpl.kt
package com.example.demo.core.usecases.impl

import com.example.demo.core.domain.Estudiante
import com.example.demo.core.port.EstudianteRepository
import com.example.demo.core.usecases.GuardarEstudiante

class GuardarEstudianteImpl(

    private val repository: EstudianteRepository

) : GuardarEstudiante {

    override fun execute(estudiante: Estudiante): Estudiante = repository.guardarEstudiante(estudiante)

}